### DISCUSSION TITLE
Api communication Bug on next project deployed on Vercel

### DISCUSSION THREAD

**Discussion by marindev1991** [vercel/community#4961] [Help]
### Question

Hello !
I have nextJS project and it is running correctly on local.
But after deployed on Vercel, I have issues.
This is in google devtools
```
message: 'Request failed with status code 500', name: 'AxiosError', code: 'ERR_BAD_RESPONSE', config: {…}, request: XMLHttpRequest
```

**Comment by amyegan** [comment]
Hey @Yakiimenko! Since a 500 status code indicates an unhandled exception, the first thing I would try is wrapping the function code in a [try...catch statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch) to log additional error info. I also recommend double checking environment variables and endpoint URLs since those are a common source of differences between development and production.

**Reply by marindev1991** [reply]
Thanks for your response.
all api were configured correctly.
Using next-auth , after login , login status is keeping but I can't read session and api calling was failing.
These are source codes related api and auth.
![Screenshot_5](https://github.com/vercel/community/assets/151897958/5d4b560f-5470-4a6b-a752-b6ea2e7a30ef)

![Screenshot_1](https://github.com/vercel/community/assets/151897958/e0b4aa07-1ab3-4914-94d7-87743f624a9d)
![Screenshot_2](https://github.com/vercel/community/assets/151897958/75409ee7-788a-4996-9e08-52f62ff3e7aa)
![Screenshot_3](https://github.com/vercel/community/assets/151897958/f442c673-6d48-48be-8ca7-6fbdf441677e)
![Screenshot_4](https://github.com/vercel/community/assets/151897958/953efa88-68b2-4057-9aa4-7fc62b28849d)

**Reply by marindev1991** [reply]
I solved the problem
